### DISCUSSION TITLE
Use custom text from query parameters

### DISCUSSION THREAD

**Discussion by aayushdutt** [monkeytypegame/monkeytype#3998] [Ideas] [answered]
Adding the ability to pass in custom text in the URL will enable creating web extensions that pass on the text to monkeytype. 

Example: An extension which lets you select the text you want to read from any website and import it into monkeytype so you can type it while reading.

**Comment by Miodec** [comment, marked as answer]
You can encode the custom text settings into the URL by pressing esc > share test settings. If you want to create such extension just check how the encoding works and creating the links should not be a problem.

**Comment by aayushdutt** [comment]
Thanks @Miodec this helped!
For anyone else looking into this - I grabbed the `compressToURI` from https://github.com/kyr0/lz-ts and used the following to get the link:

```js
const getMonkeyTypeLink = (text) => {
    const settings = JSON.stringify([
        "custom",
        null,
        {
            text: text.split(" "),
            isWordRandom: false,
            isTimeRandom: false,
            word: -1,
            time: -1,
            delimiter: " ",
        },
        null,
        null,
        null,
        null,
        null,
    ]);

    return "https://monkeytype.com/?testSettings=" + compressToURI(settings);
};
```
